source/extensions/filters/listener/http_inspector/http_inspector.cc

```diff
@@ -47,6 +46,8 @@ Network::FilterStatus Filter::onAccept(Network::ListenerFilterCallbacks& cb) {
   }
 
   cb_ = &cb;
+  config_->createStats(socket.localAddress()->asString());
```

Comment:
Do we really want to have a new set of stats for each ip address we are connecting to? I'm wondering if this will have acceptable cardinality.

Also: how often do we learn of a new local address? Is this just for an xDS event?

Reply:
Not only for the xDS event. The new local address is very common in istio's use case. Istio has a listener to captures all the traffic redirected by the iptable and selecting the filter chain according to the local address restored by the `original_dst` listener filter. e.g.,

```
listener:
  address: # address
  filter_chains:
  - filter_chain_match: # match by address and port
  - ...

  listener_filters:
  - name: envoy.listener.original_dst
  - name: envoy.listener.tls_inspector
  - name: envoy.listener.http_inspector
```

I think the stats would be more useful if we can classify the stats by the the address.

Reply:
As long as we are being mindful of the cardinality increase this incurs, and balance the cost and value.

Besides xDS, when does a new IP address get learned?

Reply:
> Besides xDS, when does a new IP address get learned?

I am not sure if I understand correctly. The address listener is listening could be different from the destination address. e.g.,  the listener is listener on `0.0.0.0:15000`, the destination address could be `1.2.3.4:80`. 

Are you asking where does the new IP address `1.2.3.4:80` get learned? Listener `0.0.0.0:15000` can specify the address `1.2.3.4` and port `80` in the filter chain match.